Sources/AuthFoundation/Token Management/Token.swift

```diff
@@ -78,13 +78,14 @@ public final class Token: Codable, Equatable, Hashable, Expires {
     
     /// Validates the claims within this JWT token, to ensure it matches the given ``OAuth2Client``.
     /// - Parameter client: Client to validate the token's claims against.
-    public func validate(using client: OAuth2Client, with context: IDTokenValidatorContext?) throws {
+    /// - Parameter issuer: The client issuer URL.
+    public func validate(using client: OAuth2Client, issuer: URL, with context: IDTokenValidatorContext?) throws {
```

Comment:
Hmm the more I think about it, I'm not sure about asking for the issuer URL since it's an extra step, and isn't needed for the majority of use-cases.  Especially since, in the future, we'll be caching the OpenIdConfiguration.

How about we make the issuer optional, with a default that falls-back to the cached openid configuration?  What are your thoughts?

Reply:
- I was thinking about it from the perspective that for an "iss" claim to be valid, the issuer url would need to be provided to `IDTokenValidator` and would need to pass the check else we through an error. 

- If we make it optional, the default (_the cached openid configuration_)is also an optional as well. Which would require changes to the   IDTokenValidator because thats a required parameter.

Also I think the validate functionality should be strictly enforced`(we can't validate if the required parameters are not provided)` 

-From my understanding, the only  case where the  openIdConfiguration( ) is not called and `issuerUrl` is not cached is only the from the idx sdk and thats why we call the  openIdConfiguration()  in parallel if its ever nil

Reply:
The more I think about it, since the `openid configuration` is cached in memory, we can also not ask for the issuer URL here.
```
    public func validate(using client: OAuth2Client, with context: IDTokenValidatorContext?) throws {
        guard let idToken = idToken, let issuer = client.openIdConfiguration?.issuer else {
            return
        }
```

However I still have concerns about not requiring the `issuer` since it's a requirement to validate a token.

Reply:
So if we unwrap the option here, and throw an appropriate error instead of silently returning, that is probably acceptable.